roomserver/internal/perform/perform_join.go

```diff
@@ -205,7 +207,14 @@ func (r *Joiner) performJoinRoomByID(
 
 	// Force a federated join if we aren't in the room and we've been
 	// given some server names to try joining by.
-	serverInRoom, _ := helpers.IsServerCurrentlyInRoom(ctx, r.DB, r.ServerName, req.RoomIDOrAlias)
+	inRoomReq := &api.QueryServerJoinedToRoomRequest{
+		RoomID: req.RoomIDOrAlias,
+	}
```

Comment:
Missing server name.

Reply:
That's acceptable — in the API comments:
```
	// Server name of the server to find. If not specified, we will
	// default to checking if the local server is joined.
```

Reply:
:S okay